assets/js/googlesitekit/datastore/user/surveys.test.js

```diff
@@ -98,6 +98,26 @@ describe( 'core/user surveys', () => {
 
 				expect( fetchMock ).not.toHaveFetched();
 			} );
+
+			it( 'should cache survey for provided ttl', async () => {
+				let ttl;
+				const triggerID = 'optimizeSurvey';
+
+				setSelectedStorageBackend( {
+					getItem: () => undefined,
+					setItem: ( _, options ) => {
+						try {
+							ttl = JSON.parse( options )?.ttl;
+						} catch {
+						}
+					},
+				} );
+
+				fetchMock.postOnce( surveyTriggerEndpoint, { body: { triggerID } } );
+
+				await registry.dispatch( STORE_NAME ).triggerSurvey( triggerID, { ttl: 500 } );
+				expect( ttl ).toBe( 500 );
```

Comment:
Elsewhere we reset the backend storage mechanism after modifying it, see https://github.com/google/site-kit-wp/blob/93d2c1b3e902cd4d91aadf82e029d0d5fabaa830/assets/js/googlesitekit/api/index.test.js#L72 for instance.

So I think we'll want to do that here too:

```suggestion
				expect( ttl ).toBe( 500 );

				// Reset the backend storage mechanism.
				setSelectedStorageBackend( undefined );
```

Reply:
Yes, makes sense to me!